proposed/2022/duplicate-nuget-item-error-handling.md

```diff
@@ -0,0 +1,142 @@
+# Duplicate NuGet item error handling
+
+- Author Name: [nkolev92](https://github.com/nkolev92)
+- Start Date 2021-04-01
+- GitHub Issue - Duplicate PackageReference [9864](https://github.com/NuGet/Home/issues/9864), CPVM duplicate erroring [9467](https://github.com/NuGet/Home/issues/9467)
+- GitHub PR [4484](https://github.com/NuGet/NuGet.Client/pull/4484)
+
+## Summary
+
+It's not uncommon that PackageReference (or PackageVersion, PackageDownload etc) items are included through secondary files and/or are manually edited, which can sometimes lead to multiple items with the same name being included.
+In some scenarios, this can cause restore inconsistencies. We will to detect these scenarios and raise a warning. NoWarn and TreatWarningsAsErrors are also respected.
+
+## Motivation
+
+Take the following project:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
```

Comment:
PackageReference Include and two PackageReference Update on same package Id/version are considered duplicates?

Reply:
Only `Include` can create a duplicate. 

You can have the following and it's valid: 

```xml
<PackageReference Include="Newtonsoft.Json" />
<PackageReference Update="Newtonsoft.Json" Version="1.0.0" />
<PackageReference Update="Newtonsoft.Json" Version="9.0.1" />
<PackageReference Update="Newtonsoft.Json" Version="12.0.3" />
```